### DISCUSSION TITLE
multiple subdomains

### DISCUSSION THREAD

**Discussion by dieterdreist** [protomaps/PMTiles#138] [Q&A] [answered]
A common trick to have the browser issue more download requests is to use multiple subdomains for the source tiles. Is this possible with pmtiles? 

Something like
```
"tiles": [
        "http://a.example.com/tiles/{z}/{x}/{y}.pbf",
        "http://b.example.com/tiles/{z}/{x}/{y}.pbf"
]
```

**Comment by bdon** [comment, marked as answer]
The domain-sharding trick is only effective for HTTP/1, and not HTTP/2. If you are using http://protomaps.com/docs/cdn the CDN edge endpoints will be using HTTP/2. 

If you are accessing PMTiles on static storage, it's possible that it uses HTTP/1 (AWS S3 is the big one). However, I don't know of a straightforward way to assign {a,b,c,d} DNS entries to an S3 bucket and that seems like a fairly niche use case.

**Comment by dieterdreist** [comment]
Thank you, I just checked and the VPS I am running is using http/3 for the client I tested, so I guess it would not make a difference for loading times if I activated sharding? Generally for low to medium traffic websites, do you think there is a benefit in running a tileserver on the webserver (e.g. have the server unpack and send individual tiles from prerendered vector tiles in an mbtiles or pmtiles file) compared to range requests on a big file?

**Reply by bdon** [reply]
Yes, even in those cases there is a benefit; one of the major drawbacks of the range-request-on-s3 approach is the latency of s3, r2 etc is poor compared to accessing "disk" like an EBS instance. Especially if you care about cache miss performance. Another drawback is that for static storage you pay per-request while a Linux server for example may only be charged based on outbound traffic. 

For many use cases the "static" nature of PMTiles is worth the extra cost and latency, but it really depends on your specific application.
